app/renderer/components/checkDefaultBrowserDialog.js

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const React = require('react')
+const ImmutableComponent = require('../../../js/components/immutableComponent')
+const Dialog = require('../../../js/components/dialog')
+const Button = require('../../../js/components/button')
+const SwitchControl = require('../../../js/components/switchControl')
+const appActions = require('../../../js/actions/appActions')
+const windowActions = require('../../../js/actions/windowActions')
+const settings = require('../../../js/constants/settings')
+
+class CheckDefaultBrowserDialog extends ImmutableComponent {
+  constructor () {
+    super()
+    this.onCheckDefaultOnStartup = this.onCheckDefaultOnStartup.bind(this)
+    this.onNotNow = this.onNotNow.bind(this)
+    this.onUseBrave = this.onUseBrave.bind(this)
+  }
+
+  onCheckDefaultOnStartup (e) {
+    windowActions.setModalDialogDetail('checkDefaultBrowserDialog', {checkDefaultOnStartup: e.target.value})
+  }
+  onNotNow () {
+    appActions.defaultBrowserUpdated(false)
+    appActions.defaultBrowserCheckComplete()
+    appActions.changeSetting(settings.CHECK_DEFAULT_ON_STARTUP, this.props.checkDefaultOnStartup)
+  }
+  onUseBrave () {
+    appActions.defaultBrowserUpdated(true)
+    appActions.defaultBrowserCheckComplete()
+    appActions.changeSetting(settings.CHECK_DEFAULT_ON_STARTUP, this.props.checkDefaultOnStartup)
```

Comment:
I wonder if all these 3 things could be managed by a single appAction instead. I won't block on it but I think that's preferred for future ref.